test/test_context.py

```diff
@@ -141,3 +142,37 @@ async def test_mite_error_backtrace_points_to_exn_source():
     assert send_fn.call_args_list[0][0][0]["location"].startswith(
         os.path.abspath(__file__)
     )
+
+
+@pytest.mark.asyncio
+async def test_transaction_name():
+    send_fn = Mock()
+    context = Context(send_fn, {})
+
+    async with context.transaction("A"):
+        assert context.transaction_name == "A"
+        async with context.transaction("B"):
+            assert context.transaction_name == "A :: B"
+            async with context.transaction("C"):
+                assert context.transaction_name == "A :: B :: C"
+            assert context.transaction_name == "A :: B"
+        assert context.transaction_name == "A"
+
+
+@pytest.mark.asyncio
+async def test_parallel_transaction_names():
+    send_fn = Mock()
+    context = Context(send_fn, {})
+
+    async def sleepy_transaction(label):
+        async with context.transaction(label):
+            assert context.transaction_name == label
+            await asyncio.sleep(1)
+            assert context.transaction_name == label
+
+    await asyncio.gather(
+        sleepy_transaction("A"),
+        sleepy_transaction("B"),
+        sleepy_transaction("C"),
+        sleepy_transaction("D"),
```

Comment:
Two transactions are sufficient to make it fail, right?  We should have the minimal test case...